addons/project/models/res_config_settings.py

```diff
@@ -11,3 +11,8 @@ class ResConfigSettings(models.TransientModel):
     module_hr_timesheet = fields.Boolean(string="Task Logs")
     group_subtask_project = fields.Boolean("Sub-tasks", implied_group="project.group_subtask_project")
     group_project_rating = fields.Boolean("Customer Ratings", implied_group='project.group_project_rating')
+    group_project_recurring_tasks = fields.Boolean("Recurring Tasks", implied_group="project.group_project_recurring_tasks")
+
+    def set_values(self):
+        self.env['project.project'].sudo().search([]).write({'allow_recurring_tasks': self.group_project_recurring_tasks})
```

Comment:
That way, we write on all the tasks on each res.config save if the option is disabled (for which the performances are already pretty bad).

https://github.com/odoo/odoo/pull/55517/files#diff-83441f41e1e80d372d7ce22c07e6900fR346 

Couldn't we check that the value has actually changed before doing that ?